src/RA_md5factory.cpp

```diff
@@ -2,65 +2,54 @@
 
 #include "RA_Defs.h"
 
-namespace {
-const static unsigned int MD5_STRING_LEN = 32;
-}
+_CONSTANT_VAR MD5_STRING_LEN = 32;
 
 std::string RAGenerateMD5(const std::string& sStringToMD5)
 {
-    static char buffer[33];
-
-    md5_state_t pms;
-    md5_byte_t digest[16];
-
-    md5_byte_t* pDataBuffer = new md5_byte_t[sStringToMD5.length()];
-    ASSERT(pDataBuffer != nullptr);
-    if (pDataBuffer == nullptr)
-        return "";
+    md5_state_t pms{};
+    std::array<md5_byte_t, 16> digest{};
 
-    memcpy(pDataBuffer, sStringToMD5.c_str(), sStringToMD5.length());
+    std::basic_string<md5_byte_t> sData;
+    for (const md5_byte_t& c : sStringToMD5)
+        sData.push_back(c);
 
     md5_init(&pms);
-    md5_append(&pms, pDataBuffer, sStringToMD5.length());
-    md5_finish(&pms, digest);
+    md5_append(&pms, sData.data(), sStringToMD5.length());
+    md5_finish(&pms, digest.data());
 
-    memset(buffer, 0, MD5_STRING_LEN + 1);
-    sprintf_s(buffer, MD5_STRING_LEN + 1,
-        "%02x%02x%02x%02x%02x%02x%02x%02x"
-        "%02x%02x%02x%02x%02x%02x%02x%02x",
-        digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
-        digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
+    std::array<char, MD5_STRING_LEN + 1> buffer{};
+    sprintf_s(buffer.data(), MD5_STRING_LEN + 1,
```

Comment:
Tried to make these safer/efficient. It seems the the md5 returned has to be on the stack.

The mem* functions shouldn't be used at all, unless we have no choice which I don't know why we wouldn't since C++ is based around RAII.